src/data/data.cu

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright 2019 by XGBoost Contributors
+ *
+ * \file data.cu
+ */
+
+#include "xgboost/data.h"
+#include "xgboost/logging.h"
+#include "xgboost/json.h"
+
+#include "columnar.h"
+#include "../common/device_helpers.cuh"
+
+namespace xgboost {
+
+void MetaInfo::SetInfo(const char * c_key, std::string const& interface_str) {
```

Comment:
This interface string is always from cuda array interface right? I think you should name variables/functions implying that it is device memory.

Reply:
@RAMitchell For now yes.  Here's a question I need to answer myself:

Interface is not designed to distinguish where is the data pointer.  So I have two options:

- Ask user to provide a `context` field in each interface object.
- Split it into two functions for host side array and device side.

Which one should I prefer.

Reply:
Doesn't cuda array interface imply the pointer always points to device memory? We are not trying to support numpys array interface here.

Note that you can always check if a pointer resides on host or device using cudaPointerGetAttributes.

Reply:
Host memory is on my to-do list, Apache Arrow is such a big family, but let's talk about it on another day.